docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,238 @@
+Calico etcd-based architecture
```

Comment:
I'm not sure, but I think this might be topsy-turvy.  The architecture, surely, is a design for components that fulfil particular roles, and the use of etcd is an internal detail.  So with that view I'd rather title this "Calico Architecture"...

Except maybe not, if we're really making a big deal of the idea that 3rd party components can orchestrate by writing directly into etcd....  In that view I guess a key property of the architecture would be having etcd as an external interface.

I'm not sure - but I think perhaps a bit more thought is needed on what we want the message to be here.  (And to take care that we're not writing in terms that reflect history (i.e. 0MQ) that will soon be forgotten.)